### ISSUE TITLE
Mina Daemon Crashes After Block Production

### ISSUE THREAD

**Issue by Makalfo** [MinaProtocol/mina#10717]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/MinaProtocol/mina/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report. Please post those things in GitHub Discussions: https://github.com/MinaProtocol/mina/discussions

### Description

The Mina Daemon crashes immediately after producing a block. It did not occur on other BPs running:

```2022-04-07 16:15:52 UTC [Info] Successfully produced a new block
2022-04-07 16:16:01 UTC [Info] Sent block with state hash $state_hash to uptime service at URL $url
	state_hash: "3NKN2YDUM9pEqXs1SdXRjndin4Jm4XSGrQevUxes84rpT6UYfFAq"
	url: "https://uptime-backend.minaprotocol.com/v1/submit"
2022-04-07 16:16:03 UTC [Info] Generated transition $state_hash was accepted into transition frontier
	state_hash: "3NKN2YDUM9pEqXs1SdXRjndin4Jm4XSGrQevUxes84rpT6UYfFAq"
2022-04-07 16:16:03 UTC [Info] Updating new available work took 10.26463508605957 ms
2022-04-07 16:16:03 UTC [Info] No more slots won in this epoch
2022-04-07 16:16:03 UTC [Info] Rebroadcasting $state_hash
	state_hash: "3NKN2YDUM9pEqXs1SdXRjndin4Jm4XSGrQevUxes84rpT6UYfFAq"
2022-04-07 16:16:03 UTC [Fatal] Unhandled top-level exception: $exn
Generating crash report
	exn: {
  "commit_id": "9b0369c27bb85c8ab2f8725c6e977eb27b53b826",
  "sexp": [
    "monitor.ml.Error",
    [
      "Hashtbl.add_exn got key already present",
      "18672045446909472430168123643691114824077651378081667266671330349930246468139"
    ],
    [
      "Raised at Base__Error.raise in file \"src/error.ml\" (inlined), line 8, characters 14-30",
      "Called from Base__Hashtbl.add_exn in file \"src/hashtbl.ml\", line 145, characters 4-21",
      "Called from Full_frontier.apply_diff in file \"src/lib/transition_frontier/full_frontier/full_frontier.ml\", line 547, characters 6-131",
      "Called from Full_frontier.apply_diffs.(fun) in file \"src/lib/transition_frontier/full_frontier/full_frontier.ml\", line 759, characters 31-74",
      "Called from Stdlib__list.fold_left in file \"list.ml\", line 121, characters 24-34",
      "Called from Base__List0.fold in file \"src/list0.ml\" (inlined), line 21, characters 22-52",
      "Called from Full_frontier.apply_diffs in file \"src/lib/transition_frontier/full_frontier/full_frontier.ml\", line 757, characters 4-484",
      "Called from Transition_frontier.add_breadcrumb_exn in file \"src/lib/transition_frontier/transition_frontier.ml\", line 348, characters 4-214",
      "Called from Transition_handler__Processor.add_and_finalize in file \"src/lib/transition_handler/processor.ml\", line 61, characters 9-67",
      "Called from Transition_handler__Processor.run.(fun) in file \"src/lib/transition_handler/processor.ml\", line 318, characters 22-122",
      "Called from Pipe_lib__Strict_pipe.Reader0.Merge.iter.read_deferred.(fun) in file \"src/lib/pipe_lib/strict_pipe.ml\", line 162, characters 30-39",
      "Called from Async_kernel__Deferred0.bind.(fun) in file \"src/deferred0.ml\", line 54, characters 64-69",
      "Called from Async_kernel__Job_queue.run_job in file \"src/job_queue.ml\" (inlined), line 128, characters 2-5",
      "Called from Async_kernel__Job_queue.run_jobs in file \"src/job_queue.ml\", line 168, characters 6-47",
      "Caught by monitor coda"
    ]
  ],
  "backtrace": [
    "Raised at Base__Error.raise in file \"src/error.ml\" (inlined), line 8, characters 14-30",
    "Called from Base__Hashtbl.add_exn in file \"src/hashtbl.ml\", line 145, characters 4-21",
    "Called from Full_frontier.apply_diff in file \"src/lib/transition_frontier/full_frontier/full_frontier.ml\", line 547, characters 6-131",
    "Called from Full_frontier.apply_diffs.(fun) in file \"src/lib/transition_frontier/full_frontier/full_frontier.ml\", line 759, characters 31-74",
    "Called from Stdlib__list.fold_left in file \"list.ml\", line 121, characters 24-34",
    "Called from Base__List0.fold in file \"src/list0.ml\" (inlined), line 21, characters 22-52",
    "Called from Full_frontier.apply_diffs in file \"src/lib/transition_frontier/full_frontier/full_frontier.ml\", line 757, characters 4-484",
    "Called from Transition_frontier.add_breadcrumb_exn in file \"src/lib/transition_frontier/transition_frontier.ml\", line 348, characters 4-214",
    "Called from Transition_handler__Processor.add_and_finalize in file \"src/lib/transition_handler/processor.ml\", line 61, characters 9-67",
    "Called from Transition_handler__Processor.run.(fun) in file \"src/lib/transition_handler/processor.ml\", line 318, characters 22-122",
    "Called from Pipe_lib__Strict_pipe.Reader0.Merge.iter.read_deferred.(fun) in file \"src/lib/pipe_lib/strict_pipe.ml\", line 162, characters 30-39",
    "Called from Async_kernel__Deferred0.bind.(fun) in file \"src/deferred0.ml\", line 54, characters 64-69",
    "Called from Async_kernel__Job_queue.run_job in file \"src/job_queue.ml\" (inlined), line 128, characters 2-5",
    "Called from Async_kernel__Job_queue.run_jobs in file \"src/job_queue.ml\", line 168, characters 6-47"
  ]
}
2022-04-07 16:16:03 UTC [Info] Saw block with state hash $state_hash
	state_hash: "3NKN2YDUM9pEqXs1SdXRjndin4Jm4XSGrQevUxes84rpT6UYfFAq"
2022-04-07 16:16:03 UTC [Fatal] Prover terminated due to signal, terminating daemon
2022-04-07 16:16:03 UTC [Info] Running async shutdown handler: "Close transition frontier, if exists"
2022-04-07 16:16:03 UTC [Info] Running async shutdown handler: "Remove daemon lockfile"
2022-04-07 16:16:03 UTC [Error] verifier terminated unexpectedly
2022-04-07 16:16:03 UTC [Info] Starting a new verifier process
2022-04-07 16:16:03 UTC [Info] verifier successfully stopped
2022-04-07 16:16:03 UTC [Fatal] libp2p_helper process died unexpectedly: "died after receiving sigkill (signal number 9)"
```

### Steps to Reproduce

1. Produce block
2. Wait for random crash 

### Expected Result

No Mina Daemon Crash

### Actual Result

Mina Daemon Crash

### How frequently do you see this issue?

Rarely

### What is the impact of this issue on your ability to run a node?

Medium

### Status

```shell
N/A as it appeared nominal until the crash
```

**Comment by p-shahi** [comment]
Seen on minaprotocol/mina-daemon:1.3.0-9b0369c-focal-mainnet

Message @Makalfo

**Comment by p-shahi** [comment]
seen on minaprotocol/mina-daemon:1.3.0-9b0369c-focal-mainnet per @Makalfo

**Comment by Makalfo** [comment]
The block was successfully produced on another mina daemon and was accepted into the canonical chain: https://minaexplorer.com/block/3NKN2YDUM9pEqXs1SdXRjndin4Jm4XSGrQevUxes84rpT6UYfFAq

**Comment by 0xthrowaway** [comment]
Same as #10164. Glad to see it finally fixed after ~2.5 months.
